File: backend/naver_link.py
"""네이버지도 딥링크 생성.

모바일에서 링크 탭 시 네이버지도 앱을 실행해 해당 주유소 경로 안내로 연결한다.
앱 미설치 시 웹 fallback URL 을 사용한다.
"""

from __future__ import annotations

from urllib.parse import quote

APP_NAME = "com.oilradar"


def app_deeplink(station_name: str) -> str:
    """네이버지도 앱 딥링크 (nmap://)."""
    return f"nmap://search?query={quote(station_name)}&appname={APP_NAME}"


def web_fallback(station_name: str) -> str:
    """앱 미설치 시 웹 fallback URL."""
    return f"https://map.naver.com/search/{quote(station_name)}"

File: frontend/src/lib/naverLink.ts
/** 네이버지도 딥링크 (백엔드 naver_link.py 와 동일 규칙). */

const APP_NAME = 'com.oilradar';

export const naverAppDeeplink = (stationName: string) =>
  `nmap://search?query=${encodeURIComponent(stationName)}&appname=${APP_NAME}`;

export const naverWebFallback = (stationName: string) =>
  `https://map.naver.com/search/${encodeURIComponent(stationName)}`;

/**
 * 모바일이면 앱 딥링크, 아니면 웹 fallback.
 * (실제 앱 실행 실패 시 fallback 전환은 추후 클릭 핸들러에서 보강)
 */
export const naverLink = (stationName: string) => naverWebFallback(stationName);
